### PR TITLE
Add new header and burger-menu - DDSDK-736

### PR DESCRIPTION
**Pull request by RoWaBo**
**What has been done:**
- New simplified header that has also been tweaked to work in the track universe. The main-menu, top-menu and search-form has been removed from the header for simplicity. Only the logo and burger-menu toggle button is visible.
- New burger menu which builds on the BUPL burger menu. It has multiple menu levels that can animate on top of each other. It combines the main menu and the top menu, displayed as a service menu styled smaller to create a better hierarchy both visually and logically.
- Top menu has been disabled at the top of the page to make it visually simpler. It's rendered in the burger menu now.
- Moved and restyled the search form from the header into the burger menu.
- Cleanup by removing CSS, JS, and templates related to the old menu and search form.
- Add icon to external links in the service menu.
- Add new fonts and colors

**Additional thought:**
I decided to reuse the BUPL burger menu to save time, but in hindsight, that might not have been a good idea since the application is different, so I needed to use some hacks to get things to work:

- The menu CSS relies on a specific HTML structure to hide and show menus. This created some restrictions when adding the search form, which made me put it inside the menu `ul`, which is not entirely semantically correct.
- In BUPL, the header that contains the burger menu is positioned relatively, whereas on DDS it's fixed. This creates some issues with making the burger menu scrollable and making sure each menu level overlaps its parent menu level.

----------------------------------------
Additional changes:
- Add nodemodules to gitignore
- Fix local npm build and watchers. Locally the assets folder was not build and fonts and images were not watched.
- Fix Make checks. Make checks failed because it tried to execute npm run clean which is no longer defined. Npm run clean used rimraf which is a node package that we no longer use or have installed.

### CODE REVIEW

File: web/themes/custom/mungo/assets_src/sass/tools/_colors.scss [hunk 1]
```diff
@@ -19,3 +21,18 @@ $pale-grey: #e6e7ea;
...
+$sand--10: #fbf9f6;
+
+// Define as css variables so they're easy to
+// override in the track universe context.
+:root {
+  --header-background-color: #{$white-one};
+  --menu-item--active: #{$light-navy};
+  --menu-item: #{$marine-blue};
+  --overlay-color: #{color.adjust($sand, $alpha: -0.3)};
+}
```
Comment (blyme) [inline web/themes/custom/mungo/assets_src/sass/tools/_colors.scss:38]: I love you

File: web/themes/custom/mungo/assets_src/sass/components/_burger-menu.scss [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    opacity: 1;
+  }
+
+  &[data-level="1"] {
+    height: 0;
+  }
+
+  &[data-level="0"] {
+    height: 100%;
+  }
```
Comment (blyme) [inline web/themes/custom/mungo/assets_src/sass/components/_burger-menu.scss:168]: Any reason this isn't added on line 156?

File: web/themes/custom/mungo/assets_src/sass/mixins/_breakpoints.mixins.scss [hunk 3]
```diff
@@ -32,3 +32,9 @@
     @content;
   }
 }
+
+@mixin media-burger-menu-large($definition: "min-width") {
+  @media ($definition: $screen-sm-min) {
+    @content;
+  }
+}
```
Comment (blyme) [inline web/themes/custom/mungo/assets_src/sass/mixins/_breakpoints.mixins.scss:40]: Doesn't line 37 prevent the parameter from having any effect? Would this make more sense?

```scss
@mixin media-burger-menu-large($definition: "min-width", $size: $screen-sm-min) {
  @media (#{$definition}: #{$size}) {
    @content;
  }
}
```

File: web/themes/custom/mungo/mungo.theme [hunk 4]
```diff
@@ -1301,3 +1299,43 @@ function mungo_theme($existing, $type, $theme, $path) {
...
+    !isset($variables["elements"][$primary_main_menu_id]["content"]["#items"]) ||
+    !isset($variables["elements"][$secondary_service_menu_id]["content"]["#items"]) ||
+    !isset($variables["elements"][$primary_track_menu_id]["content"]["#items"])
+  ) {
+    return;
+  }
+
+  // Add class to service menu items,
+  // so it can be styled different in the burger menu.
+  foreach ($variables["elements"][$secondary_service_menu_id]["content"]["#items"] as &$service_menu_item) {
```
Comment (blyme) [inline web/themes/custom/mungo/mungo.theme:1323]: Minor, but I prefer single quotes when there is no variable to interpolate

**Review by blyme — APPROVED** [review]
Good job with a quite big and annoying task :D A few comments, non-breaking. 

Other than that, in my testing I also found that on mobile, the footer menu seems to have an z-index issue. 
1. Open the menu on mobile
2. scroll all the way to the bottom.
3. You should see the white footer text overlap with some of the menu links

File: web/themes/custom/mungo/assets_src/sass/mixins/_breakpoints.mixins.scss (hunk 3 above)
Comment (RoWaBo) [reply web/themes/custom/mungo/assets_src/sass/mixins/_breakpoints.mixins.scss:40]: I understand the confusion but It does work this way.

File: web/themes/custom/mungo/mungo.theme (hunk 4 above)
Comment (RoWaBo) [reply web/themes/custom/mungo/mungo.theme:1323]: I agree

**Review by blyme — APPROVED** [review]

**Pull request merged**